src/Core/Domain/Product/Image/QueryResult/Shop/ProductImage.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Product\Image\QueryResult\Shop;
+
+class ProductImage
+{
+    /**
+     * @var int
+     */
+    private $id;
+
+    /**
+     * @var bool
+     */
+    private $isCover;
+
+    public function __construct(
+        int $id,
+        bool $isCover
+    ) {
+        $this->id = $id;
+        $this->isCover = $isCover;
+    }
+
+    /**
+     * @return int
+     */
+    public function getId(): int
```

Comment:
```suggestion
    public function getImageId(): int
```
Just a suggestion I'm not 100% sure about this one but maybe it would avoid confusion Since ProductImage is the relation between two entities (Image and Product) what does id refer to? The product or the image? That's why maybe using`getImageId` would reduce confusion

Not mandatory, just a suggestion though

Reply:
No problem